request_option_test.go

```diff
@@ -0,0 +1,100 @@
+package httpclient
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+)
+
+var errNone = errors.New("no error")
+
+func TestSetHeaders(t *testing.T) {
+	// default headers used to init the Client
+	headers := make(http.Header)
+	headers.Add("X-Test", "TestClient")
+
+	// test case struct
+	type testCase struct {
+		name     string
+		extra    map[string][]string
+		expected map[string][]string
+	}
+
+	// table tests
+	for _, tc := range []testCase{
+		{
+			name: "default case",
+			expected: map[string][]string{
+				"X-Test": {"TestClient"},
+			},
+		},
+		{
+			name: "overriding X-Test header value",
+			extra: map[string][]string{
+				"X-Test": {"Override"},
+			},
+			expected: map[string][]string{
+				"X-Test": {"Override"},
+			},
+		},
+		{
+			name: "adding X-Extra with single value",
+			extra: map[string][]string{
+				"X-Extra": {"true"},
+			},
+			expected: map[string][]string{
+				"X-Extra": {"true"},
+				"X-Test":  {"TestClient"},
+			},
+		},
+		{
+			name: "adding X-Extra with multiple values",
+			extra: map[string][]string{
+				"X-Extra": {"true", "multi"},
+			},
+			expected: map[string][]string{
+				"X-Extra": {"true", "multi"},
+				"X-Test":  {"TestClient"},
+			},
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			c, err := New(Headers(headers))
+			if err != nil {
+				t.Errorf("trouble when creating the client: %v", err)
+			}
+
+			// startup a server with handler to check that received request headers
+			// are correct
+			s := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				// ensure headers are _received_ as expected
+				for k, vs := range tc.expected {
+					exp := strings.Join(vs, ", ")
+					act := strings.Join(r.Header.Values(k), ", ")
+					if act != exp {
+						t.Errorf("expected \"%s: %s\" Header value, got \"%s: %s\"", k, exp, k, act)
+					}
+				}
+			}))
+
+			// create our extra headers
+			extra := make(http.Header)
+			for k, vs := range tc.extra {
+				for _, v := range vs {
+					extra.Add(k, v)
+				}
+			}
+			// make the request
+			err = c.Get(context.Background(), NoopResponseHandler, s.URL, SetHeaders(extra))
+			if err != nil {
+				t.Errorf("unexpected error with SetHeaders: %v", err)
+			}
+			// close client & server
+			c.Close()
+			s.Close()
```

Comment:
These could be deferred when `c` and `s` are defined instead. 

I tried making a GitHub suggestion for the change, but it couldn't compute a nice clean diff for it, so here's a patch generated from the git CLI instead!

```diff
diff --git a/request_option_test.go b/request_option_test.go
index ff56f83..9bb24ec 100644
--- a/request_option_test.go
+++ b/request_option_test.go
@@ -66,6 +66,7 @@ func TestSetHeaders(t *testing.T) {
 			if err != nil {
 				t.Errorf("trouble when creating the client: %v", err)
 			}
+			defer c.Close()
 
 			// startup a server with handler to check that received request headers
 			// are correct
@@ -79,6 +80,7 @@ func TestSetHeaders(t *testing.T) {
 					}
 				}
 			}))
+			defer s.Close()
 
 			// create our extra headers
 			extra := make(http.Header)
@@ -92,9 +94,6 @@ func TestSetHeaders(t *testing.T) {
 			if err != nil {
 				t.Errorf("unexpected error with SetHeaders: %v", err)
 			}
-			// close client & server
-			c.Close()
-			s.Close()
 		})
 	}
 }
```